react-notes/src/layouts/MainLayout/components/Sidebar/components/ArticleLink/ArticleLink.js

```diff
@@ -1,22 +1,84 @@
-import { Link } from 'react-router-dom';
+import { useContext } from 'react';
+import { Link, useNavigate, useParams } from 'react-router-dom';
 import classNames from 'classnames';
 
+import { ArticlesContext } from '../../../../../../pages/article/context';
+
 import styles from './ArticleLink.module.css';
 
-export const ArticleLink = ({ articleData }) => (
-    <li className={styles.container}>
-        <Link to={`article/${articleData.id}`}>
-            <div
-                className={classNames(
-                    articleData.title ? styles.title : styles.untitled,
-                )}
-            >
-                {articleData.title || 'Untitled'}
-            </div>
-
-            <div className={styles.lessonDate}>
-                {articleData.lessonDate || 'no date'}
-            </div>
-        </Link>
-    </li>
-);
+export const ArticleLink = ({ articleData }) => {
+    const navigate = useNavigate();
+    const { fetchArticles, sendArticles } = useContext(ArticlesContext);
+    const { articleId: currentArticleId } = useParams();
+    const articleURL = `/article/${articleData.id}`;
+
+    const onDelete = () => {
+        const articles = fetchArticles();
+        sendArticles(
+            articles.filter((article) => article.id !== articleData.id),
+        );
+
+        if (articleData.id === currentArticleId) navigate('/');
+    };
+
+    const onEdit = () => {
+        navigate(`${articleURL}/edit`);
+    };
+
+    const linkOnClickHandle = (e) => {
+        if (e.target.tagName === 'svg') e.preventDefault();
```

Comment:
The `a` tag redirect even when user click on delete and edit buttons so I needed some condition to prevent this